critiquebrainz/db/users.py

```diff
@@ -65,8 +68,570 @@ def get_many_by_mb_username(usernames):
+                   source,
+                   source_url
+              FROM review
+             WHERE user_id = :user_id
+        """), {
+            "user_id": user_id
+        })
+
+        rows = result.fetchall()
+        rows = [dict(row) for row in rows]
+
+    return rows
+
+
+def get_votes_since(user_id, date='1-1-1970'):
+    """Get votes by a user from a specified time.
+
+    Args:
+        user_id(uuid): ID of the user.
+        date(datetime): Date from which votes submitted(default: UTC).
+
+    Returns:
+        List of votes submitted by the user from the time specified
+        {
+            "vote": (bool),
+            "rated_at": (datetime)
+        }
+    """
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT vote,
+                   rated_at
+              FROM vote
+             WHERE user_id = :user_id
+               AND rated_at >= :date
+        """), {
+            "user_id": user_id,
+            "date": date
+        })
+
+        rows = result.fetchall()
+        rows = [dict(row) for row in rows]
+
+    return rows
+
+
+def get_reviews_since(user_id, date='1-1-1970'):
+    """Get reviews by a user from a specified time.
+
+    Args:
+        user_id(uuid): ID of the user.
+        date(datetime): Date from which reviews submitted(default: UTC).
+
+    Returns:
+        List of reviews by the user from the time specified
+        {
+            "id": (uuid),
+            "entity_id": (uuid),
+            "entity_type": event or place or release_group,
+            "user_id": (uuid),
+            "edits": (int),
+            "is_draft": (bool),
+            "license_id": (str),
+            "language": (str),
+            "source": (str),
+            "source_url" (str)
+        }
+    """
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT id,
+                   entity_id,
+                   entity_type,
+                   user_id,
+                   edits,
+                   is_draft,
+                   is_hidden,
+                   license_id,
+                   language,
+                   source,
+                   source_url,
+                   creation_time
+              FROM review
+         LEFT JOIN (SELECT review_id,
+                           min(timestamp) AS creation_time
+                      FROM revision
+                  GROUP BY review_id) AS review_create
+                ON review.id = review_id
+             WHERE user_id = :user_id
+               AND creation_time > :date
+        """), {
+            "user_id": user_id,
+            "date": date
+        })
+
+        rows = result.fetchall()
+        rows = [dict(row) for row in rows]
+
+    return rows
+
+
+def update(user, display_name=None, email=None, show_gravatar=None):
+    """Update info of a user.
+
+    Args:
+        user: User object whose info is to be updated.
```

Comment:
I think we shouldn't mix this stuff and just accept user's ID. You can add rows for updating dynamically depending on which arguments are specified. All of them can be merged into a dictionary of keyword arguments too.

See https://github.com/metabrainz/acousticbrainz-server/blob/4543427f492f18d89341ba190ba07eade7890753/db/dataset.py#L60-L95 for example.